msger_mosaic_bridge/include/msger_mosaic_bridge/mosaic_client.hpp

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2024 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+
+
+#ifndef MOSAIC_CLIENT_HPP
+#define MOSAIC_CLIENT_HPP
+
+#include <memory>
+#include <rclcpp/rclcpp.hpp>
+#include "conn_manager.hpp"
+
+struct ConnectionConfig {
+    std::string role_id = "";
+    std::string vehicle_id = "";
+    bool enable_registration = true;
+    bool enable_vehicle_status = true;
+
+    std::string messenger_ip_address = "127.0.0.1";
+    std::string cdasim_ip_address = "127.0.0.1"; 
+
+    unsigned short registration_port_remote = 6001;
+    unsigned short registration_port_local = 4001;
+
+    unsigned short vehicle_status_port_remote = 7001;
+    unsigned short vehicle_status_port_local = 4002;
+
+    unsigned short siren_and_light_status_port_remote = 8001;
+    unsigned short siren_and_light_status_port_local = 4003;
+};
+
+enum SirensAndLightsStatus : uint8_t
+  {
+    SIRENS_AND_LIGHTS_INACTIVE = 49,  // Value of char '1'
```

Comment:
Is there a reason this is based on char values? Is that how it's implemented somewhere else? Might not hurt to explain that if so.

Reply:
This is copied from [carma-messenger ERV plugin](https://github.com/usdot-fhwa-stol/carma-messenger/blob/bb4dcead8a71130b6b4a42352627a27dd074c30c/carma-messenger-core/emergency_response_vehicle_plugin/include/emergency_response_vehicle_plugin/emergency_response_vehicle_plugin_node.hpp#L50C4-L57C1)